### ISSUE TITLE
Validate type for members of action array parameter

### ISSUE THREAD

**Issue by Mierdin** [StackStorm/st2#3160]
@Kami brought up a good idea in review of https://github.com/StackStorm/st2/pull/3159

We don't currently validate the type of data inside an array that's passed to an action. Object parameters have this, but arrays don't. It would be cool to add this information to the action metadata, and validate the type of each member of the array.

**Comment by Kami** [comment]
Thanks for opening this and just to clarify - we do validate array items type on the API side if user specifies `items` JSON schema parameter attribute in the action metadata. For purposes of "progressive typing", it's intentionally optional.

The thing I meant is it would be good if we put some additional logic to the CLI to handle parsing and casting simple arrays with uniform non-string types when using comma-delimited string CLI notation.

For example:

```bash
param=100,200,300,400
```

With action metadata like that:

```yaml
  type: array
  items:
    type: integer
```

AFAIK, this currently wouldn't work (it would throw API validation error), because when user uses "comma-delimited string notation" we simply assume each array item is a string so we don't support arrays of integers.

In short - just a small CLI improvement :)

**Comment by Mierdin** [comment]
Ah - Thanks for the clarification. Yes, this should be fairly easy to do.
